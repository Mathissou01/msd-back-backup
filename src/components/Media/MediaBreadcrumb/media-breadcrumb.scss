@use "src/styles/02-tools/tools.typography" as typo;
@use "src/styles/02-tools/tools.buttons" as buttons;
@use "src/styles/01-settings/settings.colors" as colors;

.c-MediaBreadcrumb {
  &__Navbar {
    display: flex;
    align-items: baseline;
    padding: 8px;
    border-bottom: 1px solid colors.$gray-500;
  }

  &__NavbarItem {
    display: inline-flex;
    align-items: baseline;
  }

  &__Link {
    @include typo.text("buttonSmall");

    padding: 4px 8px;
    color: colors.$expert-blue;
    font-weight: 700;
    text-transform: capitalize;
    border-radius: 4px;

    &_active {
      color: colors.$gray-800;
      font-weight: 400;

      &:hover {
        background-color: colors.$blue-4;
      }
    }
  }

  &__Separator {
    &::after {
      margin: 0 8px;
      content: url("data:image/svg+xml,%3Csvg width='8' height='8' viewBox='0 0 6 8' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M0.833984 0.25L4.40732 3.82333C4.43054 3.84652 4.44896 3.87406 4.46153 3.90437C4.4741 3.93469 4.48057 3.96718 4.48057 4C4.48057 4.03282 4.4741 4.06531 4.46153 4.09563C4.44896 4.12594 4.43054 4.15348 4.40732 4.17667L0.833984 7.75' stroke='%23343A40' stroke-width='1.5' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E");
    }
  }

  &_displayed {
    display: none;
  }
}
