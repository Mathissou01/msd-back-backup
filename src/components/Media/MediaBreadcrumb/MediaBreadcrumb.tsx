import React from "react";
import classNames from "classnames";
import "./media-breadcrumb.scss";

export interface IMediaBreadcrumb {
  label: string;
  onClick: () => void;
}

interface IMediaBreadcrumbProps {
  foldersBreadcrumb: Array<IMediaBreadcrumb>;
}
export default function MediaBreadcrumb({
  foldersBreadcrumb,
}: IMediaBreadcrumbProps) {
  const mediaBreadcrumb = classNames("c-MediaBreadcrumb", {
    "c-MediaBreadcrumb_displayed": foldersBreadcrumb.length === 1,
  });
  return (
    <nav className={mediaBreadcrumb}>
      <ol className="c-MediaBreadcrumb__Navbar">
        {foldersBreadcrumb.map((folder, index) => (
          <li key={index} className="c-MediaBreadcrumb__NavbarItem">
            {foldersBreadcrumb.length - 1 == index ? (
              <span className="c-MediaBreadcrumb__Link">{folder.label}</span>
            ) : (
              <button
                className="c-MediaBreadcrumb__Link c-MediaBreadcrumb__Link_active"
                onClick={folder.onClick}
              >
                {folder.label}
              </button>
            )}
            {foldersBreadcrumb.length - 1 !== index && (
              <span className="c-MediaBreadcrumb__Separator" />
            )}
          </li>
        ))}
      </ol>
    </nav>
  );
}
