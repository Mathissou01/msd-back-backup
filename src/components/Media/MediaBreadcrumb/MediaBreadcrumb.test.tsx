import { render, screen } from "@testing-library/react";
import MediaBreadcrumb from "./MediaBreadcrumb";

const mockFoldersUrl = [
  {
    label: "lorem",
    onClick: jest.fn(),
  },
  {
    label: "Etiam",
    onClick: jest.fn(),
  },
  {
    label: "Vestibulum",
    onClick: jest.fn(),
  },
];

describe("MediaBreadcrumb", () => {
  it("renders", () => {
    const { container } = render(
      <MediaBreadcrumb foldersBreadcrumb={mockFoldersUrl} />,
    );
    const labelUrl = screen.getByText("lorem");
    expect(labelUrl).toMatchSnapshot();
    expect(container).toMatchSnapshot();
  });
});
