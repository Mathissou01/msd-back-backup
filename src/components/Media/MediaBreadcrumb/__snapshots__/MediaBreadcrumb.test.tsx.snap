// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaBreadcrumb renders 1`] = `
<button
  class="c-MediaBreadcrumb__Link c-MediaBreadcrumb__Link_active"
>
  lorem
</button>
`;

exports[`MediaBreadcrumb renders 2`] = `
<div>
  <nav
    class="c-MediaBreadcrumb"
  >
    <ol
      class="c-MediaBreadcrumb__Navbar"
    >
      <li
        class="c-MediaBreadcrumb__NavbarItem"
      >
        <button
          class="c-MediaBreadcrumb__Link c-MediaBreadcrumb__Link_active"
        >
          lorem
        </button>
        <span
          class="c-MediaBreadcrumb__Separator"
        />
      </li>
      <li
        class="c-MediaBreadcrumb__NavbarItem"
      >
        <button
          class="c-MediaBreadcrumb__Link c-MediaBreadcrumb__Link_active"
        >
          Etiam
        </button>
        <span
          class="c-MediaBreadcrumb__Separator"
        />
      </li>
      <li
        class="c-MediaBreadcrumb__NavbarItem"
      >
        <span
          class="c-MediaBreadcrumb__Link"
        >
          Vestibulum
        </span>
      </li>
    </ol>
  </nav>
</div>
`;
